Text/Megaparsec/Internal.hs

```diff
@@ -331,7 +331,7 @@ instance (Stream s, MonadFix m) => MonadFix (ParsecT e s m) where
           Error _ -> error "mfix ParsecT"
     runParsecT (f a) s
 
-instance MonadTrans (ParsecT e s) where
+instance Stream s => MonadTrans (ParsecT e s) where
```

Comment:
Note that the new constraint is only necessary with `transformers >= 0.6` and GHC >= 8.8:

https://hackage.haskell.org/package/transformers-0.6.0.4/docs/src/Control.Monad.Trans.Class.html#MonadTrans

For simplicity it might be best to add it unconditionally though.